std/math.d

```diff
@@ -7142,8 +7152,13 @@ bool approxEqual(T, U, V)(T lhs, U rhs, V maxRelDiff, V maxAbsDiff = 1e-5)
     {
         static if (isInputRange!U)
         {
-            // lhs is number, rhs is array
-            return approxEqual(rhs, lhs, maxRelDiff, maxAbsDiff);
+            // lhs is number, rhs is range
+            for (; !rhs.empty; rhs.popFront())
```

Comment:
The difference between `foreach` and this is that `foreach` creates a copy, this doesn't. I don't know what the implications are in terms of ranges that may fail or have weird behavior because of this. As I'm not focused on that aspect, I'd prefer to leave the implementation as is, and worry about this in a separate pull.

I will note that the original implementation does make a copy when recursing, so in this case it's not a big deal. However, it would look odd to use `foreach` here and not in the other location.